imap_processing/codice/codice_l1a.py

```diff
@@ -79,143 +84,143 @@ def __init__(self, table_id: int, plan_id: int, plan_step: int, view_id: int):
+            The 'final' ``xarray`` dataset.
         """
-        # Set the CDF attrs
-        cdf_attrs = ImapCdfAttributes()
-        cdf_attrs.add_instrument_global_attrs("codice")
-        cdf_attrs.add_instrument_variable_attrs("codice", "l1a")
-        cdf_attrs.add_global_attribute("Data_version", data_version)
-
-        # Define coordinates
-        epoch = xr.DataArray(
-            packet.epoch,
-            name="epoch",
-            dims=["epoch"],
-            attrs=cdf_attrs.get_variable_attributes("epoch"),
-        )
-        inst_az = xr.DataArray(
-            np.arange(self.num_positions),
-            name="inst_az",
-            dims=["inst_az"],
-            attrs=cdf_attrs.get_variable_attributes("inst_az"),
-        )
-        spin_sector = xr.DataArray(
-            np.arange(self.num_spin_sectors),
-            name="spin_sector",
-            dims=["spin_sector"],
-            attrs=cdf_attrs.get_variable_attributes("spin_sector"),
-        )
-        esa_step = xr.DataArray(
-            np.arange(self.num_energy_steps),
-            name="esa_step",
-            dims=["esa_step"],
-            attrs=cdf_attrs.get_variable_attributes("esa_step"),
-        )
-
-        # Create the dataset to hold the data variables
+        # Create the main dataset to hold all the variables
         dataset = xr.Dataset(
-            coords={
-                "epoch": epoch,
-                "inst_az": inst_az,
-                "spin_sector": spin_sector,
-                "esa_step": esa_step,
-            },
-            attrs=cdf_attrs.get_global_attributes(self.dataset_name),
+            coords=self.coords,
+            attrs=self.cdf_attrs.get_global_attributes(self.dataset_name),
         )
 
         # Create a data variable for each counter
         for variable_data, variable_name in zip(self.data, self.variable_names):
-            # Data arrays are structured depending on the instrument
-            if self.instrument == "lo":
-                variable_data_arr = np.array(variable_data).reshape(
-                    (
-                        1,
-                        self.num_positions,
-                        self.num_spin_sectors,
-                        self.num_energy_steps,
-                    )
-                )
-                dims = ["epoch", "inst_az", "spin_sector", "esa_step"]
-            elif self.instrument == "hi":
-                variable_data_arr = np.array(variable_data).reshape(
-                    (
-                        1,
-                        self.num_energy_steps,
-                        self.num_positions,
-                        self.num_spin_sectors,
-                    )
-                )
-                dims = ["epoch", "esa_step", "inst_az", "spin_sector"]
+            # Reshape to 4 dimensions to allow for epoch dimension
+            reshaped_variable_data = np.expand_dims(variable_data, axis=0)
```

Comment:
This doesn't seem quite right. This is adding an extra empty dimension, shouldn't we be reshaping the data you have into the proper dimensions? (Similar to what you had before I think)

Will this only ever have one epoch variable, or could there be more than that?

```python
variable_data.reshape(
                    (
                        len(dataset["epoch"],
                        self.num_energy_steps,
                        self.num_positions,
                        self.num_spin_sectors,
                    )
```

Reply:
Now that I understand this better (thank you @tech3371 for clarifying in the other comment!), you are right -- this will need to change.

I plan to make the code changes for this in a future PR that addresses #769